src/main/java/com/openshift/internal/restclient/model/DeploymentConfig.java

```diff
@@ -164,4 +188,79 @@ public IDeploymentTrigger addTrigger(String type) {
 	public String getDeploymentStrategyType() {
 		return asString(DEPLOYMENTCONFIG_STRATEGY);
 	}
+
+	@Override
+	public boolean haveTriggersFired() {
+		ModelNode causes = get(DEPLOYMENTCONFIG_CAUSES);
+		if (causes == null || causes.getType() != ModelType.LIST)
+			return false;
+		return causes.asList().size() > 0;
+	}
+
+	@Override
+	public boolean didImageTrigger(String imageTag) {
+		if (!haveTriggersFired())
+			return false;
+		ModelNode causes = get(DEPLOYMENTCONFIG_CAUSES);
+		for (ModelNode cause : causes.asList()) {
+			String type = cause.get(TYPE).asString();
+			if (type.equalsIgnoreCase(IMAGE_CHANGE)) {
+				String triggerName = cause.get(IMAGE_TRIGGER).get(FROM).get(NAME).asString();
+				if (triggerName.contains("/")) {
+					StringTokenizer st = new StringTokenizer(imageTag, "/");
```

Comment:
StringTokenizer is legacy/deprecated, highly inefficient. See [javadoc](http://docs.oracle.com/javase/7/docs/api/java/util/StringTokenizer.html)
> StringTokenizer is a legacy class that is retained for compatibility reasons although
> its use is discouraged in new code. It is recommended that anyone seeking this functionality 
> use the split method of String or the java.util.regex package instead
Please use String.split() instead,